src/vs/workbench/contrib/remote/browser/remoteIndicator.ts

```diff
@@ -547,6 +635,31 @@ export class RemoteStatusIndicator extends Disposable implements IWorkbenchContr
 		return markdownTooltip;
 	}
 
+	private async installAndRunStartCommand(metadata: RemoteExtensionMetadata) {
+		const extensionId = metadata.id;
+		const galleryExtension = (await this.extensionGalleryService.getExtensions([{ id: extensionId }], CancellationToken.None))[0];
+
+		await this.extensionManagementService.installFromGallery(galleryExtension, {
+			isMachineScoped: false,
+			donotIncludePackAndDependencies: false,
+			context: { [EXTENSION_INSTALL_SKIP_WALKTHROUGH_CONTEXT]: true }
+		});
+
+		// this.telemetryService.publicLog2<RemoteStartActionEvent, RemoteStartActionClassification>('remoteStartList.ActionExecuted', { command: 'workbench.extensions.installExtension', remoteExtensionId: extensionId });
+		await retry(async () => {
+			const ext = await this.extensionService.getExtension(metadata.id);
+			if (!ext) {
+				throw Error('Failed to find installed remote extension');
+			}
+			return ext;
+		}, 300, 10);
+
+		await this.extensionService.activateByEvent(`onCommand:${metadata.startCommand}`);
```

Comment:
Is it really necessary to do an `activateByEvent` if we're already doing `executeCommand`?

Reply:
Thanks. removed this.